data-prepper-plugins/s3-source/src/test/java/com/amazon/dataprepper/plugins/source/configuration/AwsAuthenticationOptionsTest.java

```diff
@@ -33,16 +40,68 @@ void authenticateAWSConfiguration_should_return_s3Client_without_sts_role_arn()
+        final AwsCredentialsProvider actualCredentialsProvider;
+        try (final MockedStatic<DefaultCredentialsProvider> defaultCredentialsProviderMockedStatic = mockStatic(DefaultCredentialsProvider.class)) {
+            defaultCredentialsProviderMockedStatic.when(DefaultCredentialsProvider::create)
+                    .thenReturn(mockedCredentialsProvider);
+            actualCredentialsProvider = awsAuthenticationOptions.authenticateAwsConfiguration();
+        }
+
+        assertThat(actualCredentialsProvider, sameInstance(mockedCredentialsProvider));
     }
 
-    @Test
-    void authenticateAWSConfiguration_should_return_s3Client_with_sts_role_arn() throws NoSuchFieldException, IllegalAccessException {
-        reflectivelySetField(awsAuthenticationOptions, "awsRegion", "us-east-1");
-        reflectivelySetField(awsAuthenticationOptions, "awsStsRoleArn", "arn:aws:iam::123456789012:iam-role");
-        assertThat(awsAuthenticationOptions.getAwsRegion(), equalTo("us-east-1"));
-        assertThat(awsAuthenticationOptions.getAwsStsRoleArn(), equalTo("arn:aws:iam::123456789012:iam-role"));
-        assertThat(awsAuthenticationOptions.authenticateAwsConfiguration(stsClient), instanceOf(AwsCredentialsProvider.class));
+    @Nested
+    class WithSts {
+        private StsClient stsClient;
+        private StsClientBuilder stsClientBuilder;
+
+        @BeforeEach
+        void setUp() {
+            stsClient = mock(StsClient.class);
+            stsClientBuilder = mock(StsClientBuilder.class);
+
+
+            when(stsClientBuilder.build()).thenReturn(stsClient);
+        }
+
+
+        @Test
+        void authenticateAWSConfiguration_should_return_s3Client_with_sts_role_arn() throws NoSuchFieldException, IllegalAccessException {
+            reflectivelySetField(awsAuthenticationOptions, "awsRegion", "us-east-1");
+            reflectivelySetField(awsAuthenticationOptions, "awsStsRoleArn", "arn:aws:iam::123456789012:iam-role");
+            assertThat(awsAuthenticationOptions.getAwsRegion(), equalTo("us-east-1"));
+            assertThat(awsAuthenticationOptions.getAwsStsRoleArn(), equalTo("arn:aws:iam::123456789012:iam-role"));
+
+            when(stsClientBuilder.region(Region.US_EAST_1)).thenReturn(stsClientBuilder);
+
+            final AwsCredentialsProvider actualCredentialsProvider;
+            try (final MockedStatic<StsClient> stsClientMockedStatic = mockStatic(StsClient.class)) {
+                stsClientMockedStatic.when(StsClient::builder).thenReturn(stsClientBuilder);
+                actualCredentialsProvider = awsAuthenticationOptions.authenticateAwsConfiguration();
+            }
+
+            assertThat(actualCredentialsProvider, instanceOf(AwsCredentialsProvider.class));
+        }
+
+        @Test
+        void authenticateAWSConfiguration_should_return_s3Client_with_sts_role_arn_when_no_region() throws NoSuchFieldException, IllegalAccessException {
```

Comment:
Should we have a default region when no region is specified instead of `null`?

Reply:
If `null` is supplied, the AWS SDK will supply the default. This is a [well-defined behavior](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/region-selection.html). I think if we allow for `null`, then we use this convention.

This convention is shared across AWS SDKs and the CLI.